src/components/Social/EditSocial.jsx

```diff
@@ -0,0 +1,270 @@
+  justify-content: center;
+
+  margin-left: -20px;
+
+  img {
+    width: 100%;
+    height: 100%;
+    object-fit: contain;
+  }
+`
+
+const EditProfileButton = styled(Button)`
+  font-size: 1rem;
+  box-shadow: none;
+  margin-bottom: 2rem;
+`
+
+const InputContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+  justify-content: space-between;
+  margin: 0.3rem;
+`
+
+const InputDropdown = styled(Dropdown)`
+  .react-select__control {
+    height: 2.9rem;
+    min-height: 2.5rem;
+  }
+`
+
+const InputPair = styled.div`
+  display: flex;
+  gap: 0.7rem;
+  flex-wrap: wrap;
+`
+
+const Heading = styled.div`
+  font-size: 2rem;
+  font-weight: 700;
+  color: ${p => p.theme.colors.text};
+`
+
+const SubHeading = styled.div`
+  font-size: 1.3rem;
+  font-weight: 700;
+  color: ${p => p.theme.colors.text};
+`
+
+const Input = styled.div`
+  max-width: 24rem;
+  width: 100%;
+`
+
+const InputField = styled(TextInput)`
+  width: 100%;
+  font-size: 1rem;
+  max-width: 24rem;
+  margin-left: 0 !important;
+`
+
+const Bio = styled(TextArea)`
+  font-size: 1rem;
+  width: 100%;
+  min-height: 8rem;
+`
+
+const BtnContainer = styled.div`
+  display: flex;
+`
+
+const EditSocial = ({
+  setIsEditing,
+  user,
+  userId,
+  preferredName,
+  pronouns,
+  bio,
+  role,
+  school,
+  year,
+  areaOfStudy,
+  socialLinks,
+  hideRecentlyViewed,
+  onSave,
+}) => {
+  const currentUserId = userId || user?.uid
+
+  const [name, setName] = useState(preferredName)
+  const [newPronouns, setNewPronouns] = useState(pronouns)
+  const [newBio, setNewBio] = useState(bio)
+
+  const [newRole, setNewRole] = useState(role)
+  const [newSchool, setNewSchool] = useState(school)
+  const [newYear, setNewYear] = useState(year)
+  const [newAreaOfStudy, setNewAreaOfStudy] = useState(areaOfStudy)
+
+  const [linkedin, setLinkedin] = useState(socialLinks?.linkedin ?? '')
+  const [github, setGithub] = useState(socialLinks?.github ?? '')
+  const [instagram, setInstagram] = useState(socialLinks?.instagram ?? '')
+  const [website, setWebsite] = useState(socialLinks?.website ?? '')
+  const [devpost, setDevpost] = useState(socialLinks?.devpost ?? '')
+
+  const [newHideRecentlyViewed, setNewHideRecentlyViewed] = useState(hideRecentlyViewed)
+
+  return (
+    <>
+      <TopRow>
+        <ProfilePicture>
+          <img src={veebs} alt="Profile Picture" />
+        </ProfilePicture>
+        {user?.uid === currentUserId && (
+          <BtnContainer>
+            <EditProfileButton
+              onClick={() => {
+                setIsEditing(false)
+              }}
+              color="primary"
+              width="flex"
+            >
+              Discard Changes
+            </EditProfileButton>
+            <EditProfileButton
+              disabled={name.trim() === '' || newPronouns === ''}
+              onClick={() => {
+                if (name.trim() !== '' && newPronouns !== '') {
+                  setIsEditing(false)
```

Comment:
maybe something to look into later but our `Button` component is a styled `a` instead of `button` ?? so the disabled state doesn't inherently block the onClick functionality -> so we need a check in the onClick too oop